### PR TITLE
Add prototype validation functionality

### PR DESCRIPTION
**Pull request by hayfield**
This adds prototype Schema-based validation functionality.

It is based on the previous work in this area. It aims to add basic Schema validation and validation against Codelist values.

Areas outside the core scope of this PR include tidying up resource management and validating anything more than Codelist values.

### CODE REVIEW

**Comment by dalepotter** [comment]
Tests are failing. I will fix this.

File: iati/core/resources/standard/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Resources: IATI Standard
+
+The folder (and sub-folders) contain data that is used by the library that relates to the IATI Standard.
+
+Data for a given version of the Standard is stored within a corresponding folder – for example version 2.02 data is contained with the `202` folder.  Note that this contains all content for that version of the standard, symlinks are used to ensure content replicated across multiple versions of the Standard - for example non-embedded codelists – is not replicated for each version of the Standard.
+
+Currently non-embedded codelists are stored within the `2xx/codelists_non_embedded` folder.
```
Comment (hayfield) [inline iati/core/resources/standard/README.md]: There's no `2xx` folder

File: iati/core/tests/utilities.py [hunk 2]
```diff
@@ -45,7 +72,7 @@
     'other': [NotImplemented],
     'range': [range(3, 4)],
     'set': [set(range(20)), set(['hello', 23]), frozenset(range(20)), frozenset(['hello', 23])],
-    'str': [SCHEMA_NAME_VALID, XML_STR_VALID, XML_STR_INVALID, b'\x80abc', b'\x80abc', '\N{GREEK CAPITAL LETTER DELTA}', '\u0394', '\U00000394'],
+    'str': [SCHEMA_NAME_VALID, XML_STR_VALID_NOT_IATI, XML_STR_INVALID, b'\x80abc', b'\x80abc', '\N{GREEK CAPITAL LETTER DELTA}', '\u0394', '\U00000394'],
     'tuple': [(), (1, 2)],
     'type': [type(1), type('string')],
     'unicode': [],  # counts as a string, so moed there
```
Comment (allthatilk) [inline iati/core/tests/utilities.py]: Comment needs clarity without context of commit message. Also typo 'moed' needs fixing.

File: iati/tests/test_validate.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
+"""A module containing tests for data validation."""
+import pytest
+import iati.core.data
+import iati.core.default
+import iati.core.schemas
+import iati.core.tests.utilities
+import iati.validate
+
+
+class TestValidate(object):
```
Comment (allthatilk) [inline iati/tests/test_validate.py]: If all tests are marked xfail currently, would be better to mark class as xfail instead.
https://docs.pytest.org/en/latest/skipping.html#skip-all-test-functions-of-a-class-or-module (also relevant to xfail)

Comment (hayfield) [inline iati/tests/test_validate.py]: :+1:

**Comment by allthatilk** [comment]
iati/core/tests/utilities.py could do with more clarity regarding string comments but otherwise all good.

**Pull request merged**